File: SSNodeJsChat-Android/app/src/main/java/in/newdevpoint/ssnodejschat/model/RoomResponseModel.java
package in.newdevpoint.ssnodejschat.model;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.HashMap;

public class RoomResponseModel {


    @SerializedName("roomList")
    private ArrayList<FSRoomModel> roomList = new ArrayList<>();
    @SerializedName("userList")
    private ArrayList<FSUsersModel> userList = new ArrayList<>();


    public ArrayList<FSRoomModel> getRoomList() {
        return roomList;
    }

    public void setRoomList(ArrayList<FSRoomModel> roomList) {
        this.roomList = roomList;
    }

    public ArrayList<FSUsersModel> getUserList() {
        return userList;
    }

    public void setUserList(ArrayList<FSUsersModel> userList) {
        this.userList = userList;
    }

    public HashMap<String, FSUsersModel> getUserListMap() {
        HashMap<String, FSUsersModel> tmpHashList = new HashMap<>();

        for (FSUsersModel element :
                userList) {
            tmpHashList.put(element.getId(), element);
        }
        return tmpHashList;
    }
}

File: SSNodeJsChat-Android-Kotlin/app/src/main/java/in/newdevpoint/ssnodejschat/model/RoomResponseModel.kt
package `in`.newdevpoint.ssnodejschat.model

import com.google.gson.annotations.SerializedName
import java.util.*

class RoomResponseModel {
    @SerializedName("roomList")
    var roomList = ArrayList<FSRoomModel>()

    @SerializedName("userList")
    var userList = ArrayList<FSUsersModel>()
    val userListMap: HashMap<String, FSUsersModel>
        get() {
            val tmpHashList = HashMap<String, FSUsersModel>()
            for (element in userList) {
                tmpHashList[element.id] = element
            }
            return tmpHashList
        }
}